enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,350 @@
+
+### API Extensions
+
+As part of implementing this policy, we will drop the following Godocs from [ClusterImagePolicy][ClusterImagePolicy-scope-restriction] and [ImagePolicy-scope-restriction]:
+
+```golang
+// Please be aware that the scopes should not be nested under the repositories of OpenShift Container Platform images.
+// If configured, the policies for OpenShift Container Platform repositories will not be in effect.
+```
+
+[The backing implementation][container-runtime-scope-restriction] will also be removed.
+This guard was [initially intended][scope-restriction-motivation] to protect cluster adminstrators from breaking their clusters by configuring policies that blocked critical images.
+And before Red Hat was publishing signatures for `quay.io/openshift-release-dev/ocp-release` releases, that made sense.
+But now that Red Hat is [almost][OTA-1267] publishing Sigstore signatures for those release images, it makes sense to allow policies covering those images.
+And even if a cluster administrator creates a policy that blocks critical image pulls, PodDisriptionBudgets should keep the Kubernetes API server and related core workloads running for long enough for the cluster administrator to use the Kube API to remove or adjust the problematic policy.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+HyperShift clusters run pods [on the management cluster and in the hosted cluster][hypershift-concepts].
+A ClusterImagePolicy applied to the hosted Kubernetes API server will only be distributed to hosted nodes, and none of those nodes are likely to run an image from `quay.io/openshift-release-dev/ocp-release`.
+However, a ClusterImagePolicy applied to the management cluster's Kubernetes API server would cover both the management cluster's cluster-version operator (CVO), and the CVOs associated with hosted clusters (which run as pods in mangement-cluster namespaces).
+That means that it would take an ImagePolicy to be able to distinguish between constraints on the management cluster's CVO and constraints on the hosted components running on the management cluster, which is [out of scope for this enhancement](#use-an-imagepolicy-for-openshift--namespaces).
+
+#### Standalone Clusters
+
+This proposal is aimed at standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal is aimed at single-node OpenShift deployments (SNOs), and will function there the same way it does in standalone clusters.
+It is not expected to affect resource consumption, because pod-launches are expected to be a relatively insignificant fraction of overall pod lifetime, and the `quay.io/openshift-release-dev/ocp-release` scope check covered by this proposal is expected to cheaply exclude most pods.
+
+This proposal is not relevant for MicroShift, which does not run a cluster-version operator or machine-config operator, and which handles trust via RPM signing.
+
+### Implementation Details/Notes/Constraints
```

Comment:
OKD would pull release content from `quay.io/okd/scos-release` and it would undoubtedly be signed by a different key. The release controller does this signing instead of ART.
How does the injection method proposed created the proper ClusterImagePolicy for an OKD install?

Reply:
In OpenPGP-land, it's via [two separate Dockerfiles][1], one is used in ART builds and injects ART keys, and one is used in CI builds and injects CI keys.  I suppose the machine-config operator could be set up with that same build-time divergence?  I'm not sure what the most elegant approach is; possibly there's an environment variable some kind of build script could switch on?

[1]: https://github.com/openshift/cluster-update-keys/tree/c0641e083d1bbe7d6f78e7e0c98c6d56a8b89d24

Reply:
I've pushed f2d51949 ->18deaedf adding a new section about OKD, and shifting the ClusterImagePolicy manifest from the MCO to cluster-update-keys.  I'll mark this thread resolved on that.

I'm not currently recommending an OKD policy, because I'm not sure if the OKD release controller creates Sigstore signatures, or what signing-key/Rekor it would use if/when it does.  The new approach makes it easy to layer that in via follow-up work.  But open a new thread in the new OKD or machine-update-keys section if you think we should open with an OKD policy too.